File: problem-10/c-cpp/problem-10.cpp
#include <iostream>

#include "../../utils/cpp/primes.cpp"

int main()
{

    const int SIZE = 2000000;

    // Create prime sieve
    bool *prime_sieve = newPrimeSieve(SIZE);

    long int sum = 0;

    for (long int i = 2; i < SIZE; i++)
        if (prime_sieve[i])
            sum += i;

    std::cout << sum << std::endl;

    delete prime_sieve;

    return 0;
}

File: src/com/github/problem10/problem10.java
package com.github.problem10;

import com.github.utils.Primes;

public class problem10 {
    public static void main(String[] args) {
        final int SIZE = 2000000;

        // Create prime sieve
        boolean[] prime_sieve = Primes.NewPrimeSieve(SIZE);

        long sum = 0;

        for (int i = 2; i < SIZE; i++) {
            if (prime_sieve[i]) {
                sum += (long) i;
            }
        }

        System.out.println(sum);
    }
}
